packages/date-picker/test/validation.common.js

```diff
@@ -380,36 +281,12 @@ describe('validation', () => {
       datePicker.value = '2000-01-01';
       expect(datePicker.checkValidity()).to.be.true;
     });
-
-    it('should be valid when committing a non-empty value', async () => {
```

Comment:
**note:** The new tests ensure that the component always validates _after_ the `value` property update, so it's now enough to just assert that the validation passes / fails when having / not having a value.